### DISCUSSION TITLE
Is it really worth running most tests with both of '-O(si)r' and '-O(si)'?

### DISCUSSION THREAD

**Discussion by acqn** [cc65/cc65#1816] [General]
Since most testcases don't use register variables at all, running both versions with/without `-r` looks just like a waste of time (~2x) to me.

(And yes I was only running the `-r` variants with my working copy!)

It seems to me that `-r` just need its own specific tests. Likewise `-Cl`.

**Comment by mrdudz** [comment]
So far i prefer to just run them in all configs, instead of trying to figure out which tests are relevant for what combinations - and making sure all possible side effects are still covered, now and in in the future

**Comment by acqn** [comment]
Then at least `-g -r` without `-O` is missing from "all" combinations.
